--- glide3x-20010309/h3/glide3/src/makefile.autoconf.am.orig	Tue Jul 31 12:16:45 2001
+++ glide3x-20010309/h3/glide3/src/makefile.autoconf.am	Tue Jul 31 12:16:37 2001
@@ -101,7 +101,7 @@
 endif
 
 if GL_AMD3D
-GL_AMD3D_SRC = xtexdl_3dnow.S xdraw2_3dnow.S xdraw3_3dnow.S
+GL_AMD3D_SRC = xtexdl_3dnow.S xdraw2_3dnow.S xdraw3_3dnow.S xdraw2_def.S xdraw3_def.S
 endif
 
 VDEFS = $(FX_CHRIS_DENIS_ANTHONY_HACK_VDEFS) \
@@ -138,31 +138,30 @@
 #
 # Special rules for assembly files.
 #
-xdraw2.o : xdraw2.S xdraw2.inc.S fxgasm.h
-	$(CC) $(AFLAGS) -c -o $@ xdraw2.S
+xdraw2.lo : xdraw2.S xdraw2.inc.S fxgasm.h
+	libtool --mode=compile $(CC) $(AFLAGS) -c -o $@ xdraw2.S
 
-xdraw3.o : xdraw3.S fxgasm.h
-	$(CC) $(AFLAGS) -c -o $@ xdraw3.S
+xdraw3.lo : xdraw3.S fxgasm.h
+	libtool --mode=compile $(CC) $(AFLAGS) -c -o $@ xdraw3.S
 
 # 3DNow!(tm) dependencies
 # XXX_def.obj targets are the default targets
 
 if GL_AMD3D
-else
-xdraw2_def.o: xdraw2.S xdraw2.inc.S fxgasm.h
-	$(CC) $(AFLAGS) -c -o $@ xdraw2.S
+xdraw2_def.lo: xdraw2.S xdraw2.inc.S fxgasm.h
+	libtool --mode=compile $(CC) $(AFLAGS) -c -o $@ xdraw2.S
 
-xdraw2_3dnow.o: xdraw2.S xdraw2.inc.S fxgasm.h
-	$(CC) $(AFLAGS) -DGL_AMD3D -DUSE_PACKET_FIFO=1 -c -o $@ xdraw2.S
+xdraw2_3dnow.lo: xdraw2.S xdraw2.inc.S fxgasm.h
+	libtool --mode=compile $(CC) $(AFLAGS) -DGL_AMD3D -DUSE_PACKET_FIFO=1 -c -o $@ xdraw2.S
 
-xdraw3_def.o: xdraw3.S fxgasm.h
-	$(CC) $(AFLAGS) -c -o $@ xdraw3.S
+xdraw3_def.lo: xdraw3.S fxgasm.h
+	libtool --mode=compile $(CC) $(AFLAGS) -c -o $@ xdraw3.S
 
-xdraw3_3dnow.o: xdraw3.S fxgasm.h
-	$(CC) $(AFLAGS) -DGL_AMD3D -DUSE_PACKET_FIFO=1 -c -o $@ xdraw3.S
+xdraw3_3dnow.lo: xdraw3.S fxgasm.h
+	libtool --mode=compile $(CC) $(AFLAGS) -DGL_AMD3D -DUSE_PACKET_FIFO=1 -c -o $@ xdraw3.S
 
-xtexdl_3dnow.o: xtexdl.S fxgasm.h 
-	$(CC) $(AFLAGS) -DGL_AMD3D -DUSE_PACKET_FIFO=1 -c -o $@ xtexdl.S
+xtexdl_3dnow.lo: xtexdl.S fxgasm.h 
+	libtool --mode=compile $(CC) $(AFLAGS) -DGL_AMD3D -DUSE_PACKET_FIFO=1 -c -o $@ xtexdl.S
 
 endif
 
